src/transformers/models/auto/auto_factory.py

```diff
@@ -357,6 +357,12 @@
 
 
 def _get_model_class(config, model_mapping):
+    if config.model_type not in model_mapping._config_mapping:
+        if config.model_type in model_mapping._model_mapping:
+            model_class_name = model_mapping._model_mapping[config.model_type]
+            model_class = model_mapping._load_attr_from_module(config.model_type, model_class_name)
+            return model_class
```

Comment:
To load with the model types that are not in CONFIG_MAPPING - we need a change in `SPECIAL_MODEL_TYPE_TO_MODULE_NAME` in order to use `_load_attr_from_module`.